staging/src/k8s.io/client-go/plugin/pkg/client/auth/oidc/oidc.go

```diff
@@ -120,6 +120,14 @@ func newOIDCAuthProvider(_ string, cfg map[string]string, persister restclient.A
 
 	// Check cache for existing provider.
 	if provider, ok := cache.getClient(issuer, clientID); ok {
+		if idToken := cfg[cfgIDToken]; idToken != "" {
+			// update the id token of the provider.cfg
+			provider.cfg[cfgIDToken] = idToken
+		}
+		if refreshToken := cfg[cfgRefreshToken]; refreshToken != "" {
```

Comment:
there can be multiple calls in parallel. Should nt we use the mutex to update this? I suggest writing an updateClient function.

Reply:
I add a lock here. I am checking if we should update other field.

Reply:
Please take another look